library/spdm_secured_message_lib/libspdm_secmes_encode_decode.c

```diff
@@ -397,8 +404,15 @@ libspdm_return_t libspdm_decode_secured_message(
         return LIBSPDM_STATUS_SEQUENCE_NUMBER_OVERFLOW;
     }
 
+    /* Form the AEAD IV from the salt and the sequence number. */
     libspdm_copy_mem (iv, LIBSPDM_MAX_AEAD_IV_SIZE, salt, aead_iv_size);
-    data64 = libspdm_read_uint64((const uint8_t *)iv) ^ sequence_number;
+
+    if (secured_message_context->sequence_number_endian ==
+        LIBSPDM_DATA_SESSION_SEQUENCE_NUMBER_ENDIAN_BIG) {
+        data64 = libspdm_read_uint64((const uint8_t *)iv) ^ libspdm_le_to_be_64(sequence_number);
+    } else {
+        data64 = libspdm_read_uint64((const uint8_t *)iv) ^ sequence_number;
+    }
```

Comment:
I think we need support dual_decode, like how we support dual_verify.

For encoder, choose one of little and big, like how we support asym_sign (no change is required here)